src/mame/namco/toypop.cpp

```diff
@@ -36,41 +36,47 @@
 #include "emupal.h"
 #include "screen.h"
 #include "speaker.h"
+#include "tilemap.h"
 
 
 namespace {
 
-#define MASTER_CLOCK XTAL(6'144'000)
-
 class namcos16_state : public driver_device
 {
 public:
 	namcos16_state(const machine_config &mconfig, device_type type, const char *tag) :
 		driver_device(mconfig, type, tag),
-		m_master_cpu(*this,"maincpu"),
-		m_slave_cpu(*this, "slave"),
-		m_sound_cpu(*this, "audiocpu"),
+		m_maincpu(*this,"maincpu"),
+		m_subcpu(*this, "sub"),
+		m_soundcpu(*this, "soundcpu"),
 		m_namco15xx(*this, "namco"),
 		m_namco58xx(*this, "58xx"),
 		m_namco56xx_1(*this, "56xx_1"),
 		m_namco56xx_2(*this, "56xx_2"),
 		m_palette(*this, "palette"),
 		m_gfxdecode(*this, "gfxdecode"),
-		m_master_workram(*this, "master_workram"),
-		m_slave_sharedram(*this, "slave_sharedram"),
+		m_spriteram(*this, "spriteram"),
+		m_sharedram(*this, "sharedram"),
 		m_bgvram(*this, "bgvram"),
-		m_fgvram(*this, "fgvram"),
-		m_fgattr(*this, "fgattr")
+		m_txvram(*this, "txvram"),
+		m_io_dsw(*this, "DSW%u", 1U)
 	{
 	}
 
 	void toypop(machine_config &config);
 	void liblrabl(machine_config &config);
 
+protected:
+	// driver_device overrides
+	virtual void machine_start() override ATTR_COLD;
+	virtual void machine_reset() override ATTR_COLD;
+	virtual void video_start() override ATTR_COLD;
+
 private:
-	required_device<cpu_device> m_master_cpu;
-	required_device<cpu_device> m_slave_cpu;
-	required_device<cpu_device> m_sound_cpu;
+	// devices
+	required_device<cpu_device> m_maincpu;
+	required_device<cpu_device> m_subcpu;
+	required_device<cpu_device> m_soundcpu;
 
```

Comment:
I tend to not use this kind of comment anchors: it's a repetition of what is already available with the context of MAME framework/C++ syntax, plus they are potentially prone to lag behind in case any of the struct wording changes tomorrow ...